bq/derived_table_creation/BQ_Table_Building/derived_data_views/sql/measurement_groups.sql

```diff
@@ -1,215 +1,124 @@
+  SOPClassUID,
+  cts_l0.MappingResource, -- Resource used to map the content template sequence at level 0
+  cts_l0.TemplateIdentifier, -- Unique identifier for the content template sequence at level 0
+  cs_l0.ValueType, -- Type of value associated with the content sequence at level 0
+  cs_l1_cncs.CodeMeaning, -- Code meaning associated with the concept name coding sequence at level 1
+  cs_l2.ValueType, -- Type of value associated with the content sequence at level 2
+  cs_l2_cncs.CodeValue, -- Code value associated with the concept name coding sequence at level 2
+  cs_l2_cncs.CodingSchemeDesignator, -- Coding scheme designator associated with the concept name coding sequence at level 2
+  cs_l2_cncs.CodeMeaning as cm2, -- Code meaning associated with the concept name coding sequence at level 2, with an alias of 'cm2'
+  cs_l2_rss.ReferencedSOPClassUID, -- Unique identifier for the referenced SOP class
+  cs_l2_rss.ReferencedSOPInstanceUID, -- Unique identifier for the referenced SOP instance
+  cs_l2_rss.ReferencedSegmentNumber, -- Number assigned to a referenced segment
+  cs_l2_css, -- concept Code sequence associated with the content sequence at level 2
+  cs_l1 -- Content sequence at level 1
+  FROM
+  idc-dev-etl.idc_v13_pub.dicom_metadata bid -- Data source
+  -- Left join zeroth level of ContentTemplateSequence
+  LEFT JOIN
+  UNNEST(bid.ContentTemplateSequence) cts_l0
+  -- Left join zeroth level of ContentSequence
+  LEFT JOIN
+  UNNEST(bid.ContentSequence) cs_l0
+  -- Unnest content sequence at level 1, with an offset assigned to measurementGroup_number
+  LEFT JOIN
+  unnest(cs_l0.ContentSequence) cs_l1
+  WITH
+  OFFSET
+  AS measurementGroup_number
+  -- Left join ConceptNameCodeSequence at level 1
+  LEFT JOIN
+  unnest(cs_l1.ConceptNameCodeSequence) cs_l1_cncs
+  -- Unnest content sequence at level 2
+  LEFT JOIN
+  unnest(cs_l1.ContentSequence) cs_l2
+  -- Left join ConceptNameCodeSequence at level 2
+  LEFT JOIN
+  unnest(cs_l2.ConceptNameCodeSequence) cs_l2_cncs
+  -- Left join ReferencedSOPSequence at level 2
+  LEFT JOIN
+  unnest(cs_l2.ReferencedSOPSequence) cs_l2_rss
+  -- Left join ConceptCodeSequence at level 2
+  LEFT JOIN
+  unnest(cs_l2.ConceptCodeSequence) cs_l2_css
+  WHERE
+
+  --SeriesDescription in ("BPR landmark annotations") and
+  -- We only want to include records where the TemplateIdentifier is 1500 and MappingResource is DCMR
+  TemplateIdentifier IN ('1500')
+  AND MappingResource IN ('DCMR')
+
+  -- We only want to include CONTAINER value types in the first level of content sequence
+  AND cs_l0.ValueType IN ('CONTAINER')
+
+  -- We only want to include Measurement Group Code Meanings in the second level of content sequence
+  AND cs_l1_cncs.CodeMeaning IN ("Measurement Group")
+
+  -- We want to include certain value types and code values in the third level of content sequence
+  AND (
+    -- Tracking Identifier--TEXT value type with specific Code Value and Coding Scheme Designator

```

Comment:
@vkt1414 what I suggested earlier is to include CodeMeaning value corresponding to the CodeValue/CodingSchemeDesignator values in the line below. So, in this particular example, CodeMeaning = "Tracking Identifier". Does it make sense? Can you add values of CodeMeaning corresponding to each combinations of CodeValue/CodingSchemeDesignator in this and other queries?

Reply:
Sorry, I meant mention CodeMeaning in the line above, in the comment - not in the query itself - just to make the query more readable.

Reply:
Never mind, I see I missed it. This is what happens with multitasking. It's all good!